### ISSUE TITLE
Wishlist: allow much shorter certificate lifetimes 

### ISSUE THREAD

**Issue by pronoiac** [acmesh-official/acme.sh#3875]
I've seen a CA available for self-hosting on internal instances, with ACME support. Its default certificate lifetime is 24 hours. 

From a quick look around the acme.sh source, it looks like it doesn't handle times shorter than a day. 

Is this use case something it could or should do?

**Comment by Neilpang** [comment]
>> Is this use case something it could or should do?

I'm not sure.   it depends on how many use cases are there.

**Comment by ShoootLight** [comment]
I had the same issue using Step-CA. Step-CA recommends the use passive revocation which is in fact just issueing short certificates. Passive revocation simply let certificates die. On the other side active recovation (CRL/OCSP) is often not supported by TLS implementations (see [Malone's blog entry](https://smallstep.com/blog/passive-revocation/)).

@pronoiac also when it probably too late, but the default certificate lifetime is not the maximum lifetime. Maybe it's possible to request the CA for certificates with a longer lifetime.

@Neilpang it would be nice to have an option like `--hours` (comparable to `--days`). I tried to manipulate the source code but I had not success. I manipulated the * 24-multiplicator in `--days` option inside the source code to de facto cause a calculation in hours but it just caused bugs. Also giving 0.5 as a parameter won't accept acme.sh since it is an float operation. I think the problem was located in subtracting one day but I didn't look further into it.

In the end I built a script around acme.sh powered by a cronjob. The script implements some sort of _intelligence_ by checking if the certificate reaches the end of its lifetime:
```
if openssl x509 -checkend 43200 -noout -in /path/to/fullchain.cer
then
    echo "$(date): Cert is valid for at least 12 more hours."
else
    echo "$(date): Renewing certificates"
    acme.sh --renew -d domain --force --ecc
fi
```
